### PR TITLE
Fix ANR in service launch

### PR DESCRIPTION
**Pull request by SyncedSynapse**
Fix ANRs reported on Play Store when launching the observers service.

### CODE REVIEW

File: app/src/main/java/org/xbmc/kore/service/ConnectionObserversManagerService.java [hunk 1]
```diff
@@ -67,35 +67,31 @@ public void onCreate() {
...
-        HostConnectionObserver connectionObserver = HostManager.getInstance(this).getHostConnectionObserver();
+        HostConnectionObserver connectionObserver = HostManager.getInstance(this)
+                .getHostConnectionObserver();
 
         // If we are already initialized and the same host, exit
         if (mHostConnectionObserver == connectionObserver) {
             LogUtils.LOGD(TAG, "Already initialized");
-            return START_STICKY;
+            stopForeground(true);
+            stopSelf();
```
Comment (poisdeux) [inline app/src/main/java/org/xbmc/kore/service/ConnectionObserversManagerService.java]: I fail to see the logic here. If the service is already running for the HostConnectionObserver we now stop it?

Comment (SyncedSynapse) [reply app/src/main/java/org/xbmc/kore/service/ConnectionObserversManagerService.java]: You're right. My bad. Ignore this please.

**Comment by SyncedSynapse** [comment]
@poisdeux Should be more robust now.

I'm not convinced that this will solve the ANRs reported on the Play Store, but at least we tried (the ANRs seem to be caused by something inside Android's core instead of this code). Curiously they are more prevalent on Samsung devices, so i suspect that it's one more misguided attempt of Samsung to improve battery life, with side-benefits ...

**Comment by poisdeux** [comment]
Looks good. I'm not sure why you are not convinced this will fix it. In the previous version it was possible the service was started without calling `startForeground`, resulting in the reported error. Now `startForeground` is always called.

**Pull request merged**